File: src/com/company/Cell.java
package com.company;

public class Cell {
    public int x, y;
    public Ship ship = null;
    public boolean wasShot = false;
    public boolean isShipVisible;
    private Board board;

    public Cell(int y, int x, Board board){
        this.y = y;
        this.x = x;

        this.board = board;
    }

    public boolean shoot(){
        wasShot = true;

        if (ship != null){
            ship.hit();
            if(!ship.isAlive()){
                board.ships--;
            }
            return true;
        } else {

            return false;

        }
    }
}

File: complete/src/main/resources/src/components/Cell.js
import React, {Component} from 'react'
import './style.css'

export default class Cell extends Component {
    constructor(props) {
        super(props)

        this.state = {
            x: props.x,
            y: props.y,
            isShip: props.isShip,
            isShipVisible: props.whose === 'player',
            wasShot: false,
            marker: '',
            whose: props.whose
        }
    }

    shot = () => {
        // return await fetch('', {
        //     method: 'POST',
        //     headers: 'application/json;charset=utf-8',
        //      body: JSON.stringify({
        //          'x': this.state.x,
        //          'y': this.state.y
        //      })
        // })

        this.setState(state => {
            return {
                wasShot: true,
                isShipVisible: true,
                marker: state.isShip ? 'X' : '·'
            }
        })
    }

    render() {
        return (
            <li>
                <button
                    onClick={this.shot}
                    className={`cell ${
                        this.state.isShipVisible &&
                        this.state.isShip    
                            ? 'ship' : ''
                    }`}
                >
                    {this.state.marker}
                </button>
            </li>
        )
    }
}